### PR TITLE
Upgrade telegram to v1.71

### PR DESCRIPTION
**Pull request by scribblemaniac**

### CODE REVIEW

File: Casks/telegram.rb [hunk 1]
```diff
@@ -1,14 +1,16 @@
 cask :v1 => 'telegram' do
-  version '0.8.38'
-  sha256 '1eadf4a061897070d993b2b17a208085834246548b8c6446f623e2710ec891a1'
+  version '1.71-598'
+  sha256 'aa6728944266d9c1877588d49aa83a38f3bbeb92f7f9bb5b6c64f833b2e8fa09'
 
-  # tdesktop.com is the official download host per the vendor homepage
-  url "https://updates.tdesktop.com/tmac/tsetup.#{version}.dmg"
+  # hockeyapp.net is the official download host per the vendor homepage
```
Comment (vitorgalvao) [inline Casks/telegram.rb]: Where is it?

Comment (scribblemaniac) [inline Casks/telegram.rb]: Check the appcast

Comment (vitorgalvao) [inline Casks/telegram.rb]: Figured that’d be the case. In that case, the comment must be [a bit different](https://github.com/caskroom/homebrew-cask/blob/5992783d8c95de2b0f0b1b9952847c88074daf91/Casks/slack.rb#L5).

Comment (scribblemaniac) [inline Casks/telegram.rb]: Makes sense

**Comment by scribblemaniac** [comment]
Updated

**Comment by vitorgalvao** [comment]
Can’t find this version on the website, though. Not finding the appcast, even.

**Comment by scribblemaniac** [comment]
Under the header "A native app for every platform" click "Telegram for Mac OS X". It will take you to their product on the app store ([here](
https://itunes.apple.com/us/app/messenger-for-telegram/id747648890)). You can see that there is indeed a 1.71 version. The appcast is located in that download in the standard Info.plist > SUFeedURL location.

**Comment by vitorgalvao** [comment]
But this cask isn’t for that Telegram, [but this one](https://itunes.apple.com/us/app/telegram-desktop/id946399090), that has the same name and the same vendor (yeah, not confusing at all). They also call it by different names, at times.

This is a freaking mess (and already was one, when first added). This means the appcast was pointing to the wrong one. The “main” one is shown on the website as being available only through the MAS, which will only add confusion. The problem is further exacerbated by one of them being open-sourced and the other not (I think).

At this point, I’m more inclined to just remove it outright. It’s one thing for a user to get it from the site and see, but to have this kind of confusion when getting it from us is not as acceptable. We should be clearer than this, and they don’t really let us.

Either way, this change can’t like like this, then. The cask was correct like it was except `appcast` needs to be removed.

**Comment by scribblemaniac** [comment]
This cask is far from being correct for either of the two apps. The app you are talking about is probably what this cask was intended for (based off of the version and license), however that would mean that the cask's name is incorrect as well. It is clearly called Telegram Desktop (and should have the token telegram-desktop not telegram). If we are going to use Telegram Desktop instead of Telegram, the homepage should be [here](https://desktop.telegram.org/) instead.

If we chose to go with the application I linked, we would need to make the changes in this PR and update the license.

Having said all that, I wouldn't be opposed to just scrapping this cask all together.

**Comment by vitorgalvao** [comment]
The cask’s name is correct, as it follows the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_token_reference.md). It’s a shame they use the same app bundle name for differently named apps, but it’s just one more in a series of confusing steps on their part.

What we can do is add an extra `name` (which exists for this exact reason, and was already added).

The application you linked doesn’t have a user-facing download link, so for all intents and purposes we should treat it as MAS-only.

For now I’ve corrected for the old app it was pointing to (thank you for the notice of the homepage), but will keep an eye on it, for future changes.

**Comment by scribblemaniac** [comment]
Ah I was confused with the token because the appstore's application bundle has a different name than the download we are currently using. We'll leave it at this for now.

P.S. Should casks with no user-facing download links be removed?

**Comment by vitorgalvao** [comment]
> Should casks with no user-facing download links be removed?

in general, yes, but maybe there are some that are special cases (if they are, there must be an issue somewhere discussing it, and they can be revisited).

**Pull request closed without merging**